extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/mpmetrics/AnnotationHandler.java

```diff
@@ -71,10 +71,14 @@ public void transform(TransformationContext ctx) {
                 MetricAnnotationInfo annotationInfo = new MetricAnnotationInfo(annotation, index,
                         classInfo, methodInfo, fieldInfo);
 
+                // preserve the original annotation target, `ctx.getTarget()` is different in case of method parameters
+                AnnotationInstance newAnnotation = AnnotationInstance.create(targetAnnotation, annotation.target(),
```

Comment:
Maybe rename the `targetAnnotation` param to `targetAnnotationName`? It's a bit confusing. And the same applies to `sourceAnnotation`.

Reply:
Sure

Reply:
Done.